contracts/FernBlockReward.sol

```diff
@@ -0,0 +1,43 @@
+pragma solidity ^0.4.24;
+
+import "./BlockReward.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+
+contract FernBlockReward is BlockReward {
+  using SafeERC20 for ERC20Basic;
+
+  address systemAddress;
+  ERC20Basic token;
+
+  modifier onlySystem {
+    require(msg.sender == systemAddress);
+    _;
+  }
+
+  constructor(ERC20Basic _token, address _systemAddress)
+    public
+  {
+    // systemAddress = 0xffffFFFfFFffffffffffffffFfFFFfffFFFfFFfE;
+    systemAddress = _systemAddress;
+    token = _token;
+  }
+
+  // produce rewards for the given benefactors, with corresponding reward codes.
+  // only callable by `SYSTEM_ADDRESS`
+  function reward(address[] benefactors, uint16[] kind)
```

Comment:
It'll be good to have `@param` comments above the function to clarify the role of `kind`. Maybe also use `kinds` as it is an array, similar to `benefactors`

https://solidity.readthedocs.io/en/v0.4.24/layout-of-source-files.html#comments - on syntax for `param` and other function comments

Reply:
Thanks, exactly the documentation that I was looking for! I'll add some docs now

Reply:
@mtomov added documentation to contract and function. Please take another look. Thanks.